tools/evm/T8NTool/T8NTool.cs

```diff
@@ -0,0 +1,304 @@
+            {
+                tracer.StartNewTxTrace(tx);
+                TransactionResult transactionResult = transactionProcessor.Execute(tx, new BlockExecutionContext(header), tracer);
+                tracer.EndTxTrace();
+
+                if (transactionResult.Success)
+                {
+                    successfulTxs.Add(tx);
+                    tracer.LastReceipt.PostTransactionState = null;
+                    tracer.LastReceipt.BlockHash = null;
+                    tracer.LastReceipt.BlockNumber = 0;
+                    includedTxReceipts.Add(tracer.LastReceipt);
+                } else if (transactionResult.Error != null)
+                {
+                    rejectedTxReceipts.Add(new RejectedTx(txIndex, transactionResult.Error));
+                    stateProvider.Reset();
+                }
+                stateProvider.RecalculateStateRoot();
+            }
+
+            txIndex++;
+        }
+
+        ulong gasUsed = 0;
+        if (!tracer.TxReceipts.IsNullOrEmpty())
+        {
+             gasUsed = (ulong) tracer.LastReceipt.GasUsedTotal;
+        }
+
+        Hash256 stateRoot = stateProvider.StateRoot;
+        Hash256 txRoot = TxTrie.CalculateRoot(successfulTxs.ToArray());
+        Hash256 receiptsRoot = ReceiptTrie<TxReceipt>.CalculateRoot(receiptSpec, includedTxReceipts.ToArray(), ReceiptMessageDecoder.Instance);
+
+        var postState = new PostState
+        {
+            StateRoot = stateRoot,
+            TxRoot = txRoot,
+            ReceiptRoot = receiptsRoot,
+            Receipts = includedTxReceipts.ToArray(),
+            Rejected = rejectedTxReceipts.ToArray(),
+            Difficulty = envInfo.CurrentDifficulty,
+            GasUsed = new UInt256(gasUsed),
+            CurrentBaseFee = envInfo.CurrentBaseFee,
+            WithdrawalsRoot = block.WithdrawalsRoot
+        };
+
+        var accounts = allocJson.Keys.ToDictionary(address => address, address => stateProvider.GetAccount(address));
+        foreach (Ommer ommer in envInfo.Ommers)
+        {
+            accounts.Add(ommer.Address, stateProvider.GetAccount(ommer.Address));
+        }
+        if (header.Beneficiary != null)
+        {
+            accounts.Add(header.Beneficiary, stateProvider.GetAccount(header.Beneficiary));
+        }
+        var body = Rlp.Encode(successfulTxs.ToArray()).Bytes;
+
+        return new T8NExecutionResult(postState, accounts, body);
+    }
+
+    private void WriteToFile(string filename, string? basedir, object outputObject)
+    {
+        FileInfo fileInfo = new(basedir + filename);
+        Directory.CreateDirectory(fileInfo.DirectoryName!);
+        using StreamWriter writer = new(fileInfo.FullName);
+        writer.Write(_ethereumJsonSerializer.Serialize(outputObject, true));
+    }
+
+    private bool IsPostMerge(IReleaseSpec spec)
+    {
+        return spec == ArrowGlacier.Instance
+               || spec == GrayGlacier.Instance
+               || spec == Paris.Instance
```

Comment:
Glaciers are pre-merge
```suggestion
        return spec == Paris.Instance
```